helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -453,6 +462,18 @@ public static Object getGenericDefaultValue(Schema.Field field) {
     return ADAPTER.getGenericDefaultValue(field);
   }
 
+  /**
+   * returns a new schema field, cloned from an existing schema field and a given schema.
+   * @param field a schema field
+   * @param schema a schema
+   * @return a new field that cloned from the given field and schema
+   * @throws org.apache.avro.AvroRuntimeException if the field in question has no default.
+   */
+  public static Schema.Field cloneSchemaField(Schema.Field field, Schema schema) {
```

Comment:
i think a cleaner API would be:
`public static FieldBuilder cloneField(Schema.Field original)`
users get a builder, and can override any properties they like and call build()

Reply:
I thought for avro-utils, we should not return the "internal" class but return a type in AVRO libraries, like Schema, or Decoder etc.